### PR TITLE
Implement timeseries zoom drill thru

### PR DESCRIPTION
**Pull request by camsaul**
Implements #33556

There are a couple of nice-to-haves that I'll probably address once I get some feedback from @bshepherdson to understand if I'm even on the right track here

- [ ] Probably a good idea not to show the `hour` and `minute` drills for `:type/Date` columns since those would result in invalid queries, but I guess that could be done as a follow-on. (#33573)
- [ ] We shouldn't suggest a drill thru if adding it would result in a duplicate breakout on the same column. Duplicate breakouts are against the rules in MBQL; see https://github.com/metabase/metabase/issues/32489. Again, MLv1/FE doesn't do this, but we probably should do it. This is generally not really a problem on the FE however because it does not let you add multiple breakouts on the same column

### CODE REVIEW

File: src/metabase/lib/drill_thru/zoom_in_timeseries.cljc [hunk 1]
```diff
@@ -1,32 +1,94 @@
...
+                  breakouts)))
+
+(mu/defn ^:private next-breakout-unit :- [:maybe ::lib.schema.temporal-bucketing/unit.date-time.truncate]
+  [column :- lib.metadata/ColumnMetadata]
+  (when-let [current-unit (lib.temporal-bucket/raw-temporal-bucket column)]
+    (when (contains? (set valid-current-units) current-unit)
+      (loop [[unit & [next-unit :as more]] valid-current-units]
+        (cond
+          (= unit current-unit) next-unit
+          (seq more)            (recur more))))))
```
Comment (bshepherdson) [inline src/metabase/lib/drill_thru/zoom_in_timeseries.cljc:52]: Would this be neater as a map?
```clojure
(def next-unit
  (zipmap (drop-last 1 valid-current-units) (drop 1 valid-current-units)))
```

File: src/metabase/lib/drill_thru/zoom_in_timeseries.cljc [hunk 2]
```diff
@@ -1,32 +1,94 @@
...
+       :value        value
+       :next-unit    next-unit})))
+
+(mu/defmethod lib.drill-thru.common/drill-thru-method :drill-thru/zoom-in.timeseries
+  [query                            :- ::lib.schema/query
+   stage-number                     :- :int
+   {:keys [column value next-unit]} :- ::lib.schema.drill-thru/drill-thru.zoom-in.timeseries]
+  (let [breakout     (matching-breakout-ref query stage-number column)
+        new-breakout (lib.temporal-bucket/with-temporal-bucket breakout next-unit)]
+    (as-> query query
```
Comment (bshepherdson) [inline src/metabase/lib/drill_thru/zoom_in_timeseries.cljc:92]: This can just be `->`

**Review by bshepherdson — APPROVED** [review]
This looks good to me. It's based on top of some clearly substantial refactoring of the core of drills that I haven't reviewed, though. (It looks like mostly splitting each drill into its own namespace, and maybe fixing the dispatch function for `drill-thru-method`; not sure if there's much else?)

File: src/metabase/lib/drill_thru/zoom_in_timeseries.cljc (hunk 2 above)
Comment (camsaul) [reply src/metabase/lib/drill_thru/zoom_in_timeseries.cljc]: good call. Fixed

File: src/metabase/lib/drill_thru/zoom_in_timeseries.cljc (hunk 1 above)
Comment (camsaul) [reply src/metabase/lib/drill_thru/zoom_in_timeseries.cljc]: good call. Fixed

**Pull request merged**